### PR TITLE
BI-12138 shopping basket limit error

### PR DESCRIPTION
**Pull request by aokunade**
BI-12138 Shopping Basket basket limit error

### CODE REVIEW

File: src/controllers/basket.controller.ts [hunk 1]
```diff
@@ -146,3 +162,23 @@ const deliverableItemsHaveAddressCheck = (basketResource: Basket): boolean => {
...
+/**
+ * displayBasketLimitError controls the presentation of a basket limit warning/error as appropriate.
+ * @return whether a basket limit error is to be displayed (<code>true</code>), or not (<code>false</code>)
+ */
+ const displayBasketLimitError = (req: Request,
+    res: Response,
+    basketLimit: BasketLimit) : boolean => {
+if (basketLimit.basketLimitState == BasketLimitState.BELOW_LIMIT) {
+    const nextPage = `${CHS_URL}`;
+logger.debug(`Basket is not full, redirecting to ${nextPage}`);
```
Comment (VaughanJackson123) [inline src/controllers/basket.controller.ts:176]: I think the indentation is a bit strange in this method.

**Comment by VaughanJackson123** [comment]
I think we need to remove the `.idea` directory from this PR, and sort out that indentation issue, then we should be good to go. 👍

**Comment by caismith-ch** [comment]
I don't think the `.idea` files should be committed to the repository as they are user specific. Can they be removed and an entry for `.idea` be added to the `.gitignore` file please.

File: src/controllers/basket.controller.ts (hunk 1 above)
Comment (aokunade) [reply src/controllers/basket.controller.ts:176]: Do you mean it should be the first one I changed from?

**Comment by aokunade** [comment]
> I don't think the `.idea` files should be committed to the repository as they are user specific. Can they be removed and an entry for `.idea` be added to the `.gitignore` file please.

It must have been committed in error, will resolve it now.

File: src/controllers/basket.controller.ts (hunk 1 above)
Comment (VaughanJackson123) [reply src/controllers/basket.controller.ts:176]: Not sure about that. But the logging/implemention lines within the `if` branch here should be indented to reflect the fact they are within the branch. Same for the lines making up the `else` branch.

**Review by VaughanJackson123 — APPROVED** [review]

**Pull request merged**